### DISCUSSION TITLE
CPU Training

### DISCUSSION THREAD

**Discussion by SimplAiDev** [d8ahazard/sd_dreambooth_extension#773] [Q&A] [answered]
hi,
I updated dreambooth_extension and after that i cant see CPU Training option anymore.
is the CPU Training option removed from the extension ?

I was doing CPU training for my models.

Thanks

**Comment by runner22k** [comment]
How log it used to take per model? How many images you used, per training?

**Reply by SimplAiDev** [reply]
I have AMD 8320 FX CPU with 4 CORE that takes 9 hours to train 4 images of 512x512 Res.

**Reply by runner22k** [reply]
My goodness! too much for 4 images! my be try Google colab at some point, and use LoRA. I was also looking for dreambooth options for 8GBVRAM. Later I got convinced to relay on LoRA instead of entire model tuning. If we have a 16GB VRAM card, right now it will take 8 minutes to traing 72 images, (6 images per batch), through Google colab, if we get a faster GPU, I tried 3 times and I got lucky once, else it takes 2 hours on google colab.

I agree that dreambooth training is easy as LoRA needs captions (that is the headache I need to deal with to get the best out of LoRA training.)

**Comment by MaKraMc** [comment]
I have the same problem. Sadly my GPU is not powerful enough but my CPU could do the training over night.

**Reply by SimplAiDev** [reply]
yes i was doing my models on CPU overnight. it is not fast as GPU but works really well.

but i am not sure if it is me only or CPU training options is being removed ?

**Reply by MaKraMc** [reply]
I didn't find the option as well.

**Comment by MaKraMc** [comment]
I really don't know why it has been removed. I did some digging in an older version and reimplemented it into the code. Will be doing some testing tomorrow but it should be pretty straightforward. :)
![grafik](https://user-images.githubusercontent.com/69326651/213552792-0be22871-4f81-4b2b-888f-73f2f60f09ec.png)

**Reply by SimplAiDev** [reply]
oh that would be great. i am waiting for the result.

**Reply by make23l** [reply]
I've been looking for this option for two days! It's good to know that it still exists even though it has been lost. I'm waiting! thx!

**Reply by gsgoldma** [reply]
please post the code, if you have got it figured out.

**Comment by d8ahazard** [comment, marked as answer]
Add --force-cpu to your command line arguments and restart. This will force
CPU only training.

**Reply by MaKraMc** [reply]
Will give that a try. Thanks. Why did the option get removed in the first place?

**Reply by d8ahazard** [reply]
> Will give that a try. Thanks. Why did the option get removed in the first place?

Because with the implementations of LORA and split text encoder/unet training, it should be possible to train without needing CPU on most machines that at least support CUDA. Because it's so much slower, I was trying to nudge people towards other options...but I forgot about people without Nvidia GPUs; and so added it back as a commandline arg. This way, it's still an option, but not one that's so readily available.

**Reply by SimplAiDev** [reply]
thank you so much i will try this

**Reply by MaKraMc** [reply]
Yeah, thanks for reaching out and for your awesome work so far.

**Reply by SimplAiDev** [reply]
I did try to add **--force-cpu** inside **webui-user.bat** and also inside **webui-user-dreambooth.bat** but no success maybe i am doing something wrong.

if i add **--force-cpu** inside **webui-user.bat** i got error in which it says NO ARGUMENT EXIST.
if i add **--force-cpu** inside **webui-user-dreambooth.bat** i got error in which it gives me OOM Error and VRAM FULL Error.

Thank You for your help

**Reply by gsgoldma** [reply]
> > Will give that a try. Thanks. Why did the option get removed in the first place?
> 
> Because with the implementations of LORA and split text encoder/unet training, it should be possible to train without needing CPU on most machines that at least support CUDA. Because it's so much slower, I was trying to nudge people towards other options...but I forgot about people without Nvidia GPUs; and so added it back as a commandline arg. This way, it's still an option, but not one that's so readily available.

Will this will force the cpu for the whole program and not just dreambooth training?

**Reply by d8ahazard** [reply]
Just for Dreambooth

**Reply by SimplAiDev** [reply]
**Thank you so much** Now it is working. I am able to train on CPU again.

I have one more question my CPU usage don't go more than **60%** while training is there something wrong with it or it just meant to do like this?

Thank you Once Again

**Reply by econundrum** [reply]
> > > Will give that a try. Thanks. Why did the option get removed in the first place?
> > 
> > 
> > Because with the implementations of LORA and split text encoder/unet training, it should be possible to train without needing CPU on most machines that at least support CUDA. Because it's so much slower, I was trying to nudge people towards other options...but I forgot about people without Nvidia GPUs; and so added it back as a commandline arg. This way, it's still an option, but not one that's so readily available.
> 
> Will this will force the cpu for the whole program and not just dreambooth training?

So is it possible to train using these options on 4Gb vRAM? should I just add --force-cpu to the dreambooth bat?

**Reply by SimplAiDev** [reply]
yes it is possible if you add --force-cpu to the dreambooth bat.Rather than using GPU and 4Gb VRam it will use CPU and RAM for Training.This process is really slow.

**Reply by econundrum** [reply]
I know it's slow I trained a few things on the older version, I don't seem to be able to get --force-cpu to work CUDA is still trying to allocate vRAM from the error I'm getting.

**Reply by SimplAiDev** [reply]
I got same problem too but after one or two restarts it worked properly.

only just to make sure did you add that line in weui-user.bat file inside sd-dreambooth folder?

please check attached image for more information
![Screenshot_1](https://user-images.githubusercontent.com/86287028/219468861-3c4ba785-99a0-4504-b153-d7e8ab1e22de.png)
.

**Reply by gsgoldma** [reply]
> I got same problem too but after one or two restarts it worked properly.
> 
> only just to make sure did you add that line in weui-user.bat file inside sd-dreambooth folder?
> 
> please check attached image for more information ![Screenshot_1](https://user-images.githubusercontent.com/86287028/219468861-3c4ba785-99a0-4504-b153-d7e8ab1e22de.png) .

which one should it be added to? I added to the extension dream booths folder commandline args, is that wrong?

**Reply by econundrum** [reply]
Thank you I'd added it in the wrong place I'll give that a try.

**Comment by econundrum** [comment]
Have to say colab which I've tried now is a much better option than cpu training generally, still the option is nice.
